ntp-proto/src/packet/extensionfields.rs

```diff
@@ -300,27 +297,39 @@ impl<'a> ExtensionField<'a> {
         {
             return Err(std::io::ErrorKind::WriteZero.into());
         }
+
+        // move the ciphertext over to make space for nonce padding
         cur_extension_field.copy_within(
             header_size + nonce_length..header_size + nonce_length + ciphertext_length,
             header_size + padded_nonce_length,
         );
-        cur_extension_field[header_size + nonce_length..header_size + padded_nonce_length]
-            .copy_from_slice(&padding[..padded_nonce_length - nonce_length]);
-        cur_extension_field[header_size + padded_nonce_length + ciphertext_length
-            ..header_size + padded_nonce_length + padded_ciphertext_length]
-            .copy_from_slice(&padding[..padded_nonce_length - nonce_length]);
 
-        // Final header
+        // zero out then nonce padding
+        let nonce_padding = padded_nonce_length - nonce_length;
+        cur_extension_field[header_size + nonce_length..][..nonce_padding]
+            .copy_from_slice(&padding[..nonce_padding]);
+
+        // zero out the ciphertext padding
+        let ciphertext_padding = padded_ciphertext_length - ciphertext_length;
+        debug_assert_eq!(
+            ciphertext_padding, 0,
+            "extension field encoding should add padding"
+        );
```

Comment:
I believe this just should always be zero in practice. We don't test with a `ciphertext_padding > 0` at the moment, and I don't think it's possible to create with our logic a plaintext that would require ciphertext padding

Reply:
Given the multiple-of-4 requirement even on encrypted efs you might be right. I am comfortable with this debug assert for now.